docs/content/develop/run-tests.md

```diff
@@ -1,53 +1,186 @@
+
+    `-run` is regexp-like, so multiple tests can be run in parallel by specifying a common substring of those tests (for example, `TestAccContainerNodePool` to run all node pool tests).
+
+1. Optional: Save verbose test output to a file for analysis.
+
+    ```bash
+    TF_LOG=TRACE make testacc TEST=./google TESTARGS='-run=TestAccContainerNodePool_basic' > output.log
+    ```
+
+1. Optional: debug tests with [Delve](https://github.com/go-delve/delve):
+
+    ```bash
+    ## Navigate to the google package within your local GCP Terraform provider Git clone.
+    cd $GOPATH/src/github.com/terraform-providers/terraform-provider-google/google
+
+    ## Execute the dlv command to launch the test.
+    ## Note that the --test.run flag uses the same regexp matching as go test --run.
+    TF_ACC=1 dlv test -- --test.v --test.run TestAccComputeRegionBackendService_withCdnPolicy
+    ```
+
+{{< /tab >}}
+
+{{< tab "Beta Provider" >}}
+
+1. Run unit tests and linters
+
+    ```bash
+    cd $GOPATH/src/github.com/hashicorp/terraform-provider-google-beta
+    make test
+    make lint
+    ```
+
+1. Run acceptance tests
+
+    ```bash
+    make testacc TEST=./google-beta TESTARGS='-run=TestAccContainerNodePool_basic'
+    ```
+
+    TESTARGS allows you to pass [testing flags](https://pkg.go.dev/cmd/go#hdr-Testing_flags) to `go test`. The most important is `-run`, which allows you to limit the tests that get run. There are 2000+ tests, and running all of them takes over 9 hours and requires a lot of GCP quota.
+
+    `-run` is regexp-like, so multiple tests can be run in parallel by specifying a common substring of those tests (for example, `TestAccContainerNodePool` to run all node pool tests).
+
+1. Optional: Save verbose test output to a file for analysis.
+
+    ```bash
+    TF_LOG=TRACE make testacc TEST=./google-beta TESTARGS='-run=TestAccContainerNodePool_basic' > output.log
+    ```
+
+1. Optional: debug tests with [Delve](https://github.com/go-delve/delve):
+
+    ```bash
+    ## Navigate to the google package within your local GCP Terraform provider Git clone.
+    cd $GOPATH/src/github.com/terraform-providers/terraform-provider-google-beta/google-beta
+
+    ## Execute the dlv command to launch the test.
+    ## Note that the --test.run flag uses the same regexp matching as go test --run.
+    TF_ACC=1 dlv test -- --test.v --test.run TestAccComputeRegionBackendService_withCdnPolicy
+    ```
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+### Testing with different `terraform` versions
+
+Tests will use whatever version of the `terraform` binary is found on your path. To test with multiple versions of `terraform` core, you must run the tests multiple times with different versions. You can use [`tfenv`](https://github.com/tfutils/tfenv) to manage your system `terraform` versions.
+
+## Test manually
```

Comment:
Please rewrite this section to be more procedural and provide less context. Possibly something like the following subsections:

- Set up test environment
   - Could include setting up developer overrides or filesystem mirrors
- Run tests
  - Could include building the provider, creating a test directory and test configurations, and running `terraform apply` with debug output
- Cleanup
   - Could include any teardown for developer overrides or filesystem mirrors.

For filesystem mirrors - like we talked about, it's okay to leave that content less procedural for now since it's not the default path. However, developer overrides should be procedural, with less context overall - in particular, we should only recommend one path for using them, either global configuration or an environment variable. (I'd have a slight preference for an environment variable that they pass directly in copy-pasteable commands but am open to other options as long as there is a single recommended path.)

Reply:
```suggestion
## Optional: Test manually
```

The developer overrides section is looking better, but it would be great to format this in sections like:

- Setup
- Run manual tests
  - This would include building the provider
- Cleanup

The main changes for this are:

1. Move the provider build step into the "Using Terraform CLI developer overrides" section
2. Update section headers & move them out of the tabs (so they show up in the sidebar nav)
   - This will mean setting up tabs in each section & adding content for the "Run manual tests" section for filesystem mirrors.